pkg/cfn/builder/vpc.go

```diff
@@ -35,20 +42,52 @@ func (c *ClusterResourceSet) addSubnets(refRT *gfn.Value, topology api.SubnetTop
 			SubnetId:     refSubnet,
 			RouteTableId: refRT,
 		})
+
+		c.newResource(alias+"CIDRv6", &gfn.AWSEC2SubnetCidrBlock{
+			SubnetId:      refSubnet,
+			Ipv6CidrBlock: getCIDRv6(counter),
+		})
+
 		c.subnets[topology] = append(c.subnets[topology], refSubnet)
+		counter++
 	}
 }
 
+var refCIDRv6 = gfn.MakeFnCIDR(
+	gfn.MakeFnSelect(0, gfn.MakeFnGetAttString("VPC.Ipv6CidrBlocks")), 8, 64,
+)
+
+func getCIDRv6(n int) *gfn.Value {
+	return gfn.MakeFnSelect(n, refCIDRv6)
+}
+
 //nolint:interfacer
 func (c *ClusterResourceSet) addResourcesForVPC() {
 	internetCIDR := gfn.NewString("0.0.0.0/0")
+	//internetCIDRv6 := gfn.NewString("::/0")
```

Comment:
remember to remove the comment before merging (and the others about DestinationIpv6CidrBlock)